### PR TITLE
Enable pkg/cpuid support on arm64.

### LINKED ISSUE #1255: cpuid is required for Arm64
cpuid is required for Arm64

### PR DESCRIPTION
**Pull request by xiaobo55x**
Fixes #1255

Signed-off-by: Haibo Xu <haibo.xu@arm.com>
Change-Id: I8614e6f3ee321c2989567e4e712aa8f28cc9db14

### CODE REVIEW

**Comment by xiaobo55x** [comment]
Ref #63

**Comment by xiaobo55x** [comment]
Hi @fvoznika, @prattmic,

Please help review this patch!

Thanks,
Haibo

File: pkg/cpuid/cpuid_arm64_unsafe.go [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+
+// ARM64 doesn't have a 'cpuid' equivalent, which means it have no architected
+// discovery mechanism for hardware features available to userspace code at EL0.
+// The kernel exposes the presence of these features to userspace through a set
+// of flags(HWCAP/HWCAP2) bits, exposed in the auxilliary vector.
+// Ref Documentation/arm64/elf_hwcaps.rst for more info.
+//
+// Currently, only the HWCAP bits are supported.
+
+const (
```
Comment (avagin) [inline pkg/cpuid/cpuid_arm64_unsafe.go]: Unsafe file has to contain only unsafe code and nothing extra. I think all code form this file except initHwCap should be moved in pkg/cpuid/cpuid_arm64.go.

Comment (xiaobo55x) [reply pkg/cpuid/cpuid_arm64_unsafe.go]: Done. Thanks!

File: pkg/cpuid/cpuid_arm64_unsafe.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+func initHwCap() {
+	auxv, err := ioutil.ReadFile("/proc/self/auxv")
+	if err != nil {
+		log.Warningf("Could not read /proc/self/auxv: %v", err)
+		return
+	}
+
+	l := len(auxv) / 16
+	for i := 0; i < l; i++ {
+		tag := *(*uint64)(unsafe.Pointer(&auxv[i*16]))
```
Comment (prattmic) [inline pkg/cpuid/cpuid_arm64_unsafe.go]: This doesn't need to be unsafe. Just use binary.LittleEndian.Uint64 to parse the bytes.

File: pkg/cpuid/cpuid_arm64_unsafe.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+import (
+	"io/ioutil"
+	"unsafe"
+
+	"gvisor.dev/gvisor/pkg/log"
+)
+
+// The auxiliary vector of a process on the Linux system can be read from /proc/self/auxv,
+// and tags and values are stored as 8-bytes decimal key-value pairs on the 64-bit system.
```
Comment (prattmic) [inline pkg/cpuid/cpuid_arm64_unsafe.go]: Minor nit: please wrap comments to 80 columns.

File: pkg/cpuid/cpuid_arm64_unsafe.go (hunk 2 above)
Comment (xiaobo55x) [reply pkg/cpuid/cpuid_arm64_unsafe.go]: Done. Thanks!

File: pkg/cpuid/cpuid_arm64_unsafe.go (hunk 3 above)
Comment (xiaobo55x) [reply pkg/cpuid/cpuid_arm64_unsafe.go]: Done. Thanks!

**Comment by majek** [comment]
bump

**Pull request merged**